### ISSUE TITLE
Why Application Gateway is a resource under Network?

### ISSUE THREAD

**Issue by pinaki-accenture** [hashicorp/terraform-provider-azurerm#15459]
While all other services like 'LoadBalancer', 'Logic Apps' are direct modules under 'azurerm', 'Application gateway' still is a resource under 'Network'.

The problem is unlike 'load balancer' all resources for app gateway doesn't work as individual resource and we have to create a lot of dynamic blocks. Especially 'rewrite_rule_set' becomes an insane block and creating a variable for this block with validation is really painful.

```
dynamic "rewrite_rule_set" {
    for_each = var.appgw_rewrite_rule_set
    content {
      name = lookup(rewrite_rule_set.value, "name", null)
      dynamic "rewrite_rule" {
        for_each = lookup(rewrite_rule_set.value, "rewrite_rule", null)
        content {
          name          = lookup(rewrite_rule.value, "name", null)
          rule_sequence = lookup(rewrite_rule.value, "rule_sequence", null)

          dynamic "condition" {
            for_each = lookup(rewrite_rule.value, "condition", {})
            content {
              ignore_case = lookup(condition.value, "condition_ignore_case", null)
              negate      = lookup(condition.value, "condition_negate", null)
              pattern     = lookup(condition.value, "condition_pattern", null)
              variable    = lookup(condition.value, "condition_variable", null)
            }
          }

          dynamic "response_header_configuration" {
            for_each = lookup(rewrite_rule.value, "response_header_name", null) != null ? ["fake"] : []
            content {
              header_name  = lookup(rewrite_rule.value, "response_header_name", null)
              header_value = lookup(rewrite_rule.value, "response_header_value", null)
            }
          }

          dynamic "request_header_configuration" {
            for_each = lookup(rewrite_rule.value, "request_header_name", null) != null ? ["fake"] : []
            content {
              header_name  = lookup(rewrite_rule.value, "request_header_name", null)
              header_value = lookup(rewrite_rule.value, "request_header_value", null)
            }
          }

          dynamic "url" {
            for_each = lookup(rewrite_rule.value, "url_reroute", null) != null ? ["fake"] : []
            content {
              path         = lookup(rewrite_rule.value, "url_path", null)
              query_string = lookup(rewrite_rule.value, "url_query_string", null)
              reroute      = lookup(rewrite_rule.value, "url_reroute", null)
            }
          }
        }
      }
    }
  }
```

**Comment by tombuildsstuff** [comment]
hi @pinaki-accenture 

Thanks for opening this issue.

When provisioning an Application Gateway unfortunately it's required to create at least one of most of the sub-components, as such it's not possible to add support for these as separate resources without introducing a perectual diff on the main `azurerm_application_gateway` resource. This has come up a number of times over the years and is unfortunately blocked on the Azure Service Team who need to add support for this - which is currently being tracked [in this upstream issue](https://github.com/Azure/azure-rest-api-specs/issues/2313).

Unfortunately the Azure Networking team have since tried to move this Feature Request to the Azure Feedback site, which has since been shut down - as such [we've previously asked for an update](https://github.com/Azure/azure-rest-api-specs/issues/2313#issuecomment-997844834) to get this re-opened - however it may also be worth opening a new issue on that repository too.

Since the API doesn't allow provisioning an Application Gateway without any subcomponents at the moment unfortunately and so is a feature request for the Azure Service Team - I'm going to close this issue for the moment, but I'd recommend either commenting on the existing issue above, or making a new one so that the Service Team can take a look - but unfortunately there's not much we can do from Terraform's side at the moment.

Thanks!
